src/handlers/generate-payout-permit.ts

```diff
@@ -1,123 +1,35 @@
-import { PermitTransactionData } from "../types/permits";
+import { Permit } from "../types/permits";
 import { Context } from "../types/context";
 import { generateErc20PermitSignature } from "./generate-erc20-permit";
 import { generateErc721PermitSignature } from "./generate-erc721-permit";
-import { getLabelsFromLinkedIssue, getPriceFromLabels, getWalletRecord, handleNoWalletFound, unpackInputs } from "../utils/helpers";
+import { PermitRequest } from "../types/plugin-input";
 
 /**
  * Generates a payout permit based on the provided context.
  * @param context - The context object containing the configuration and payload.
  * @returns A Promise that resolves to the generated permit transaction data or an error message.
  */
-export async function generatePayoutPermit(context: Context): Promise<PermitTransactionData | string> {
-  const { isNftRewardEnabled } = context.config;
-  const logger = context.logger;
-  const eventName = context.eventName;
-
-  if (eventName == "pull_request.closed") {
-    const payload = context.payload as Context<"pull_request.closed">["payload"];
-    return await generatePayoutForPullRequest(context, payload, isNftRewardEnabled);
-  } else if (eventName == "workflow_dispatch") {
-    return await generatePayoutForWorkflowDispatch(context, isNftRewardEnabled);
-  } else {
-    logger.error("Invalid payload");
-    return "Permit not generated: invalid payload";
-  }
-}
-
-/**
- * Generates a payout permit from a workflow dispatch.
- * @notice All inputs must be passed in from the previous plugin/kernel.
- */
-export async function generatePayoutForWorkflowDispatch(context: Context, isNftRewardEnabled: boolean): Promise<PermitTransactionData | string> {
-  const inputs = unpackInputs(context);
-  const logger = context.logger;
-
-  let permit: PermitTransactionData | string;
-
-  if (inputs.erc20) {
-    if (!inputs.erc20.token || !inputs.erc20.amount || !inputs.erc20.spender || !inputs.erc20.networkId) {
-      logger.error("No token, amount, spender, or networkId found for ERC20 permit");
-      return "Permit not generated: no token, amount, spender, or networkId found for ERC20 permit";
+export async function generatePayoutPermit(context: Context, permitRequests: PermitRequest[]): Promise<Permit[]> {
+  const permits: Permit[] = [];
+
+  for (const permitRequest of permitRequests) {
+    const { type, amount, username, contributionType } = permitRequest;
+
+    let permit: Permit;
+    switch (type) {
```

Comment:
Why switch case for this

Reply:
I think it's fine, if eventually we want to have more permits? It's always better to have a switch case than an if 🌲 forest.

Reply:
We don't have any plans for different types of permits